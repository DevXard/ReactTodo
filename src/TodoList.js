import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import { useState } from 'react';
import {v4 as uuid} from 'uuid'


const TodoList = () => {
    const [data, setData] = useState([])
    
    const addNewTodo = (todo) => {
        setData(t => ([...t, {todo, id: uuid()}]))
        
    }

    const removeTodo = (id) => {
        setData(data => data.filter(todo => todo.id !== id))
    }
    return (
        <div>
            <h2>Todo List</h2>
            <NewTodoForm addNewTodo={addNewTodo} />
            {data.map((todo) => <Todo id={todo.id} removeTodo={removeTodo} key={todo.id} todo={todo.todo} />)}
        </div>
    )
}

export default TodoList;