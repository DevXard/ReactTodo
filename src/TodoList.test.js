import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("should render without crashing", () => {
    render(<TodoList />)
})

it("matches the snapshot", () => {
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot()
})

it("Should add new item", () => {
    const { queryByText, getByLabelText } = render(<TodoList />)
    const input = getByLabelText("Todo")
    const btn = queryByText("Add Todo")
    expect(queryByText("Get Milk")).not.toBeInTheDocument()

    fireEvent.change(input, {target: {value: "Get Milk"}})
    fireEvent.click(btn);
    expect(queryByText("Get Milk")).toBeInTheDocument()
})