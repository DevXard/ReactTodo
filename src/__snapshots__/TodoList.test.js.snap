// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
